README.md

```diff
@@ -48,95 +53,117 @@ defmodule MyProject.Mixfile do
+```
 
-To use the client, simply call `start_link`, supplying the host and port.
+We generate two versions of each function defined by the Thrift service's
+interface: one that returns a standard result tuple, and a `!` variant that
+returns a single result value but raises an exception if an error occurs.
 
 ```elixir
-iex> alias Thrift.Test.UserService.Binary.Framed.Client
-iex> {:ok, client} = Client.start_link("localhost", 2345, [])
-iex> {:ok, user} = Client.get_user_by_id(client, 22451)
-{:ok, %Thrift.Test.User{id: 22451, username: "stinky", first_name: "Stinky", last_name: "Stinkman"}}
+@spec add(pid(), integer(), integer(), Client.options()) :: {:ok, integer()} | {:error, any()}
+def add(client, left, right, rpc_opts \\ [])
+
+@spec add!(pid(), integer(), integer(), Client.options()) :: integer()
+def add!(client, left, right, rpc_opts \\ [])
 ```
 
-## Using The Server
-Creating a thrift server is slightly more involved than creating the client, because
-you need to create a module to handle the work. Fortunately, Elixir Thrift
-creates a [Behaviour](http://elixir-lang.org/docs/stable/elixir/behaviours.html#content),
-complete with correct success typing, for this module. To implement this behaviour,
-use the `@behaviour` module attribute. The compiler will now inform you about
-any missed functions.
+## Servers
 
-Here is an implementation for the server defined above:
+In order to start a Thrift server, you will need to provide a callback module
+that implements the functions described by its service interface. Fortunately,
+a [behaviour][] module will be automatically generated for you, complete with
+success typing.
 
 ```elixir
-defmodule UserServiceHandler do
-  @behaviour Thrift.Test.UserService.Handler
-
-  def ping, do: true
-
-  def get_user_by_id(user_id) do
-    case Backend.find_user_by_id(user_id) do
-      {:ok, user} ->
-        user
-      {:error, _} ->
-        raise Thrift.Test.UserNotFound.exception message: "could not find user with id #{user_id}"
-    end
-  end
+defmodule CalculatorHandler do
+  @behaviour Calcualtor.Generated.Service.Handler
 
-  def delete_user(user_id) do
-    Backend.delete_user(user_id) == :ok
+  @impl true
+  def add(left, right) do
+    left + right
   end
 end
 ```
 
-To start a server with UserServiceHandler as the callback module:
+Then provide your handler module when starting the server process:
 
 ```elixir
-{:ok, server_pid} = Thrift.Test.UserService.Binary.Framed.Server.start_link(UserServiceHandler, 2345, [])
+iex> alias Calculator.Generated.Service.Binary.Framed.Server
+iex> {:ok, server} = Server.start_link(CalculatorHandler, 9090, [])
 ```
 
-...and your server is up and running. RPC calls to the server are delegated to UserServiceHandler.
+All RPC calls to the server will be delegated to the handler module. The server
+provides a [supervisor][] which can be added to your application's supervision
+tree. It's important to start it as a `supervisor` and not as a `worker`.
```

Comment:
Maybe make this more explicit by saying:
``It's important to add it to your supervision tree with type `:supervisor` and not `:worker`.``